executor/aggfuncs/builder.go

```diff
@@ -135,6 +137,46 @@ func buildApproxCountDistinct(aggFuncDesc *aggregation.AggFuncDesc, ordinal int)
 	return nil
 }
 
+func buildApproxPercentile(sctx sessionctx.Context, aggFuncDesc *aggregation.AggFuncDesc, ordinal int) AggFunc {
+	if aggFuncDesc.Mode == aggregation.DedupMode {
+		return nil
+	}
+
+	percent, isNull, err := expression.GetIntFromConstant(sctx, aggFuncDesc.Args[1])
```

Comment:
We should move the parameters check to planner, where build logical aggregation.

Reply:
I have moved it into `inferType4ApproxPercentile`.

I'll push it later.